toolchain/check/handle_pattern_list.cpp

```diff
@@ -37,6 +37,12 @@ auto HandleParseNode(Context& context, Parse::TuplePatternStartId node_id)
   context.node_stack().Push(node_id);
   context.param_and_arg_refs_stack().Push();
   context.BeginSubpattern();
+  // TODO: Remove this branch once the parse tree differentiates between
+  // tuple patterns and param patterns.
+  if (context.full_pattern_stack().CurrentKind() ==
+      FullPatternStack::Kind::ImplicitParamList) {
```

Comment:
Can the stack be in different states than ImplicitParamList here? How does that happen?